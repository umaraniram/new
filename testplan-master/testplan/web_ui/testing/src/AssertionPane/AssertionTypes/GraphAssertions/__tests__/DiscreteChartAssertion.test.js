import React from 'react';
import {shallow} from 'enzyme';
import {StyleSheetTestUtils} from "aphrodite";

import DiscreteChartAssertion from '../DiscreteChartAssertion.js';

function defaultProps() {
  return {
    assertion: {
                "meta_type": "entry",
                "utc_time": "2019-07-12T12:36:09.796381+00:00",
                "machine_time": "2019-07-12T12:36:09.796381+00:00",
                "type": "DiscreteChart",
                "series_options": {
                    "Data Name": {
                        "colour": "literal"
                    }
                },
                "line_no": 101,
                "graph_data": {
                    "Data Name": [{
                            "angle": 1,
                            "color": "#89DAC1",
                            "name": "green"
                        }, {
                            "angle": 2,
                            "color": "#F6D18A",
                            "name": "yellow"
                        }, {
                            "angle": 5,
                            "color": "#1E96BE",
                            "name": "cyan"
                        }, {
                            "angle": 3,
                            "color": "#DA70BF",
                            "name": "magenta"
                        }, {
                            "angle": 5,
                            "color": "#F6D18A",
                            "name": "yellow again"
                        }
                    ]
                },
                "description": "Pie Chart",
                "category": "DEFAULT",
                "graph_options": null,
                "graph_type": "Pie"
            }
    };
}


describe('DiscreteChartAssertion', () => {
  let props;
  let shallowComponent;

  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
    props = defaultProps();
    shallowComponent = undefined;
  });

  it('can render basic markup without error', () => {
    shallow(<DiscreteChartAssertion {...props}/>).html();
  });

  it('shallow renders the correct HTML structure', () => {
    shallowComponent = shallow(<DiscreteChartAssertion {...props}/>);
    expect(shallowComponent).toMatchSnapshot();
  });

});