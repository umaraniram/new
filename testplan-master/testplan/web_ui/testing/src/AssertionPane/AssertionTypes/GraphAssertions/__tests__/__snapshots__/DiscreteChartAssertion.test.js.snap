// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscreteChartAssertion shallow renders the correct HTML structure 1`] = `
<div>
  <RadialChart
    className=""
    colorRange={
      Array [
        "#12939A",
        "#79C7E3",
        "#1A3177",
        "#FF9833",
        "#EF5D28",
      ]
    }
    colorType="literal"
    data={
      Array [
        Object {
          "angle": 1,
          "color": "#89DAC1",
          "name": "green",
        },
        Object {
          "angle": 2,
          "color": "#F6D18A",
          "name": "yellow",
        },
        Object {
          "angle": 5,
          "color": "#1E96BE",
          "name": "cyan",
        },
        Object {
          "angle": 3,
          "color": "#DA70BF",
          "name": "magenta",
        },
        Object {
          "angle": 5,
          "color": "#F6D18A",
          "name": "yellow again",
        },
      ]
    }
    getAngle={[Function]}
    getAngle0={[Function]}
    getLabel={[Function]}
    getRadius={[Function]}
    getRadius0={[Function]}
    getSubLabel={[Function]}
    height={300}
    key="Data Name"
    onSeriesMouseOut={[Function]}
    onValueMouseOver={[Function]}
    padAngle={0}
    width={400}
  />
  <DiscreteColorLegendItem
    className=""
    colors={
      Array [
        "#12939A",
        "#79C7E3",
        "#1A3177",
        "#FF9833",
        "#EF5D28",
      ]
    }
    items={
      Array [
        Object {
          "color": "#89DAC1",
          "title": "green",
        },
        Object {
          "color": "#F6D18A",
          "title": "yellow",
        },
        Object {
          "color": "#1E96BE",
          "title": "cyan",
        },
        Object {
          "color": "#DA70BF",
          "title": "magenta",
        },
        Object {
          "color": "#F6D18A",
          "title": "yellow again",
        },
      ]
    }
    orientation="horizontal"
    width={750}
  />
  <br />
  <p>
    (Hover over chart to see labels)
  </p>
</div>
`;
